.github/workflows/ibis-ci.yml

```diff
@@ -47,6 +47,21 @@ jobs:
           cache: 'poetry'
       - name: Install dependencies
         run: poetry install
+      - name: Cache Cargo
+        uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cargo/bin/
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+            ibis-server/rust/target/
+          key: ${{ runner.os }}-cargo-${{ hashFiles('ibis-server/rust/Cargo.lock') }}
+      - name: Test rust code
```

Comment:
We need to have test for different platform (macos, mac-arm, linux, windows) but I think we don't need to do that in this PR. Have another follow up PR is better.

Reply:
Let's adjust it in the new PR to adjust the layout in the future.